src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs

```diff
@@ -130,24 +147,36 @@ public INetworkPeer CreateNetworkPeer(TcpClient client, NetworkPeerConnectionPar
                 Services = NetworkPeerServices.Nothing,
             };
 
-            return await this.CreateConnectedNetworkPeerAsync(ipEndPoint, parameters);
+            return await this.CreateConnectedNetworkPeerAsync(ipEndPoint, parameters, connectedPeersHelper);
         }
 
         /// <inheritdoc/>
-        public async Task<INetworkPeer> CreateConnectedNetworkPeerAsync(IPEndPoint peerEndPoint, NetworkPeerConnectionParameters parameters = null)
+        public async Task<INetworkPeer> CreateConnectedNetworkPeerAsync(
+            IPEndPoint peerEndPoint,
+            NetworkPeerConnectionParameters parameters = null,
+            ConnectedPeersHelper connectedPeersHelper = null)
         {
             Guard.NotNull(peerEndPoint, nameof(peerEndPoint));
 
-            var peer = new NetworkPeer(peerEndPoint, this.network, parameters, this, this.dateTimeProvider, this.loggerFactory);
+            Action<INetworkPeer> onDisconnected = null;
+
+            if (connectedPeersHelper != null)
```

Comment:
This is a bit of a hack, as I suggested earlier we can inject the PeerLifetime (or what ever we call it maybe PeerDisposer) directly to the peer itself.

Reply:
There's a fundamental misunderstanding of the purpose of this component, let's have Skype call for that.